src/test/scala/ai/privado/languageEngine/java/audit/DataFlowReportTest.scala

```diff
@@ -0,0 +1,161 @@
+package ai.privado.languageEngine.java.audit
+
+import ai.privado.audit.DataFlowReport
+import ai.privado.cache.AuditCache
+import ai.privado.cache.AuditCache.SourcePathInfo
+import ai.privado.dataflow.Dataflow
+import ai.privado.entrypoint.PrivadoInput
+import ai.privado.languageEngine.java.audit.TestData.AuditTestClassData
+import ai.privado.languageEngine.java.tagger.source.IdentifierTagger
+import ai.privado.entrypoint.ScanProcessor
+import ai.privado.tagger.sink.RegularSinkTagger
+import io.joern.dataflowengineoss.layers.dataflows.{OssDataFlow, OssDataFlowOptions}
+import io.shiftleft.semanticcpg.layers.LayerCreatorContext
+
+import scala.collection.mutable
+
+class DataFlowReportTest extends DataFlowReportTestBase {
+
+  override val javaFileContentMap: Map[String, String] = getContent()
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    val privadoInput = PrivadoInput(
+      Set.empty,
+      Set.empty,
+      Set.empty,
+      Set.empty,
+      false,
+      false,
+      false,
+      false,
+      false,
+      false,
+      false,
+      false,
+      false,
+      false,
+      false,
+      false,
+      false,
+      false,
+      false,
+      false
+    )
+
+    ScanProcessor.config = privadoInput
```

Comment:
@ankit-privado This should not be done as, it can cause the same problem which we have observed with RuleCache, instead the place where it is throwing you the error `privadoInput` needs to be made available there

Reply:
Done, using `privadoScanConfig` instance inside dataFlow instead of calling `ScanProcessor.config`.